### PR TITLE
Fix a bug where modifiers' effects continue after they're removed

### PR DESCRIPTION
**Pull request by swankjesse**
---

- [x] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/__Snapshots__/UIViewFlexContainerTestHost/testColumnWithChildModifierChanges.Empty.png
Comment (swankjesse) [inline redwood-layout-uiview/RedwoodLayoutUIViewTests/__Snapshots__/UIViewFlexContainerTestHost/testColumnWithChildModifierChanges.Empty.png:1]: We already had a test that demonstrated that our behavior was incorrect!

File: build-support/src/main/resources/app/cash/redwood/buildsupport/flexboxHelpers.kt [hunk 1]
```diff
@@ -85,59 +85,120 @@ internal fun CrossAxisAlignment.toAlignSelf() = when (this) {
...
+  if (newMarginTop neq oldMarginTop) marginTop = newMarginTop
+  if (newMarginBottom neq oldMarginBottom) marginBottom = newMarginBottom
+  if (newAlignSelf != oldAlignSelf) alignSelf = newAlignSelf
+  if (newRequestedMinWidth neq oldRequestedMinWidth) requestedMinWidth = newRequestedMinWidth
+  if (newRequestedMaxWidth neq oldRequestedMaxWidth) requestedMaxWidth = newRequestedMaxWidth
+  if (newRequestedMinHeight neq oldRequestedMinHeight) requestedMinHeight = newRequestedMinHeight
+  if (newRequestedMaxHeight neq oldRequestedMaxHeight) requestedMaxHeight = newRequestedMaxHeight
+  if (newFlexGrow neq oldFlexGrow) flexGrow = newFlexGrow
+  if (newFlexShrink neq oldFlexShrink) flexShrink = newFlexShrink
+  if (newFlexBasis neq oldFlexBasis) flexBasis = newFlexBasis
```
Comment (swankjesse) [inline build-support/src/main/resources/app/cash/redwood/buildsupport/flexboxHelpers.kt:195]: https://www.reddit.com/r/ProgrammerHumor/comments/13gt6co/standagainstfloats/

**Review by colinrtwhite — APPROVED** [review]
Good catch!

**Review by JakeWharton — APPROVED** [review]

**Comment by JakeWharton** [comment]
Linked from modifier removal detection issue. No one should have to do this.

**Pull request merged**